FrontEnd/src/BookingPage/BookingPage.jsx

```diff
@@ -0,0 +1,126 @@
+import "./BookingPage.css";
+import { useLocation } from "react-router-dom";
+import { useState } from "react";
+import { requestAppointment } from "./BookingDB";
+import Modal from "../Modal";
+import NotificationsPage from "../Notifications/NotificationsPage";
+
+function BookingForm({ userData }) {
+  const location = useLocation();
+  const { post, userUID } = location.state || {};
+  const invisibleComponent = false;
+  const [isModalShown, setIsModalShown] = useState(false);
+  const [appointmentData, setAppointmentData] = useState({
+    appointmentTitle: "",
+    appointmentDescription: "",
+    appointmentDate: "",
+    appointmentTime: "",
+    appointmentAdditionalNote: "",
+    vendorUsername: post.vendorUsername,
+  });
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setAppointmentData({ ...appointmentData, [name]: value });
+  };
+
+  const sendAppointmentRequest = async (
+    userUID,
+    vendorUID,
+    appointmentData,
+    userData
+  ) => {
+    await requestAppointment(userUID, vendorUID, appointmentData, userData);
+  };
+
+  const handleSubmit = async (event) => {
+    try {
+      event.preventDefault();
+      await sendAppointmentRequest(
+        userUID,
+        post.userId,
+        appointmentData,
+        userData
+      );
+      toggleModal();
```

Comment:
I suggest you add `isRequestPending` state. Set it to `true` when request is sent and to `false` when a response is received. While it's pending, prevent the modal from closing.
Otherwise if a user closes the modal before the response comes back, this `toggleModal` call will reopen it.